# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/AudioConduit.cpp
# Commit: 069c5baf6d32
# Full Hash: 069c5baf6d3240279a8322605e1fe799738121c9
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 3
# Description:
#   Bug 1654112 - Implement AudioConduit::DeleteStreams. r=ng
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D103493
# ==============================================================================

diff -r 06ae2b9a2703 -r 069c5baf6d32 dom/media/webrtc/libwebrtcglue/AudioConduit.cpp
--- a/dom/media/webrtc/libwebrtcglue/AudioConduit.cpp	Fri Jan 29 15:51:59 2021 +0100
+++ b/dom/media/webrtc/libwebrtcglue/AudioConduit.cpp	Thu Jan 28 10:54:32 2021 +0100
@@ -55,15 +55,21 @@
   return MakeRefPtr<WebrtcAudioConduit>(aCall, aStsThread);
 }
 
+void WebrtcAudioConduit::DeleteStreams() {
+  MOZ_ASSERT(NS_IsMainThread());
+  MutexAutoLock lock(mMutex);
+  DeleteSendStream();
+  DeleteRecvStream();
+}
+
 /**
  * Destruction defines for our super-classes
  */
 WebrtcAudioConduit::~WebrtcAudioConduit() {
   CSFLogDebug(LOGTAG, "%s ", __FUNCTION__);
   MOZ_ASSERT(NS_IsMainThread());
-  MutexAutoLock lock(mMutex);
-  DeleteSendStream();
-  DeleteRecvStream();
+  MOZ_ASSERT(!mSendStream && !mRecvStream,
+             "Call DeleteStreams prior to ~WebrtcAudioConduit.");
 }
 
 bool WebrtcAudioConduit::SetLocalSSRCs(const std::vector<uint32_t>& aSSRCs,